### ISSUE TITLE
Internal Server Error: `rsession did not start in time`

### ISSUE THREAD

**Issue by barrymoo** [jupyterhub/nbrsessionproxy#36]
Hello All,

This package looks like exactly what I need! However, I am having an issue where I get a 500 Internal Server Error when hitting `/user/{user}/rstudio` that says `rsession did not start in time`. I have the `rstudio-server` service running and followed the installation instructions. Unfortunately, I don't know where to go from here. Any help is appreciated.

Thanks.

**Comment by ryanlovett** [comment]
Are you starting rstudio-server via systemd, upstart, or some other init system? If so, nbrsessionproxy is currently incompatible with that. It expects that the notebook user can invoke `rserver --www-port={some_port}` which will want to write to /tmp/rstudio-server as the notebook user. If the init system has started rstudio-server on its own as a system user, the notebook user won't be able to write there. You can verify this by manually running the above rserver command (perhaps `/usr/lib/rstudio-server/bin/rserver --www-port={some_port}`).

This also means that [only one user](https://github.com/jupyterhub/nbrsessionproxy/issues/33) can launch rstudio-server on a multi-user system if /tmp/ hasn't been namespaced.

**Comment by barrymoo** [comment]
I see. I am starting via systemd. This is helpful.

Multi-user environment would be great.

Closing.

**Comment by ryanlovett** [comment]
Yes, I've used pam_namespace to namespace /tmp; I just looked but couldn't find other methods. It looks like the remote user authenticator doesn't do anything with respect to PAM so I think you'd need to run a custom authenticator that takes bits from or subclasses PAMAuthenticator and the remote user authenticator. pam_namespace runs from the session module type and that is handled by [pre_spawn_start](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/auth.py#L666).
